src/mount/mount-tool.c

```diff
@@ -372,51 +409,54 @@ static int parse_argv(int argc, char *argv[]) {
                         return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                                "At most two arguments required.");
 
-                if (arg_mount_type && !fstype_is_blockdev_backed(arg_mount_type)) {
-                        arg_mount_what = strdup(argv[optind]);
+                if (arg_tmpfs) {
+                        arg_mount_what = strdup("tmpfs");
```

Comment:
if two args are specified, then the first arg should be used as `what` and the 2nd as `where`.

right now you'll entirely ignore the 2nd arg, but allow it

Reply:
Yes, but it doesn't make sense to have a 'what' with --tmpfs, no?

Reply:
why not? the kernel takes the string and shows it in the mount table. Might be good for decoration to give an fs a name beyond the path it is mounted to.

Reply:
Added a check that enforces only a single argument with --tmpfs

Reply:
> why not? the kernel takes the string and shows it in the mount table. Might be good for decoration to give an fs a name beyond the path it is mounted to.

isn't it supposed to be 'tmpfs' always? that's how I always used it